code/modules/mining/equipment/kinetic_crusher.dm

```diff
@@ -113,6 +113,8 @@
 		return
 	if(!proximity_flag && charged)//Mark a target, or mine a tile.
 		var/turf/proj_turf = user.loc
+		if(ismecha(user.loc))
```

Comment:
this only ever applies to the subtype. make a small proc for getting the origin of the projectile, and override it in the subtype.